### PR TITLE
A few initiative things

### PR DESCRIPTION
**Pull request by jdevnull**
1. Fix for wound modifiers applying twice to initiative
2. Add messaging for the case where a character rolls zero or negative initiative (only possible through a low roll + wound penalties).

These two address: https://discord.com/channels/564618629467996170/788953927269613608/1218251123249647756

3. With the now available delta grade move-by-wires cyberware, they are now feasible for character builds, but they were missing a key feature. MBW3 and MBW4, by the book, are supposed to gain extra actions in the first one or two passes through an initiative turn. We can simulate that by adding 10/20 to their initiative roll, though this is mechanically weaker because the extra actions will occur at the end of an initiative turn instead of near the beginning.

### CODE REVIEW

File: src/fight.cpp [hunk 1]
```diff
@@ -5705,18 +5705,37 @@ void roll_individual_initiative(struct char_data *ch)
...
       GET_INIT_ROLL(ch) -= 10;
       AFF_FLAGS(ch).RemoveBit(AFF_ACTION);
     }
+
+    // By the book, MBW 3 adds an extra action to their first pass and 4 adds an additional action to their second.
+    // This implementation is mechanically weaker since chars will get the extras at the end of the turn.
+    for (obj_data *cyber = ch->cyberware; cyber; cyber = cyber->next_content) {
+      if (GET_CYBERWARE_TYPE(cyber) == CYB_MOVEBYWIRE) {
+        if (GET_CYBERWARE_RATING(cyber) == 3)
+          GET_INIT_ROLL(ch) += 10;
```
Comment (luciensadi) [inline src/fight.cpp:5721]: +10 is a full extra pass, how about +5 for 3 and +10 for 4?

File: src/fight.cpp [hunk 2]
```diff
@@ -5728,7 +5747,7 @@ void roll_individual_initiative(struct char_data *ch)
         && !MOB_FLAGGED(FIGHTING(ch), MOB_INANIMATE)
         && GET_MOBALERT(FIGHTING(ch)) == MALERT_CALM
         && success_test(GET_REA(ch), 4) > success_test(GET_REA(FIGHTING(ch)), 4)) {
-      GET_INIT_ROLL(FIGHTING(ch)) = 0;
+      GET_INIT_ROLL(FIGHTING(ch)) <= 0;
```
Comment (luciensadi) [inline src/fight.cpp:5750]: Typo? This changed from an assignment to a no-op comparison.

Comment (jdevnull) [reply src/fight.cpp:5750]: Oops, yeah, not sure how that happened.

File: src/fight.cpp (hunk 1 above)
Comment (jdevnull) [reply src/fight.cpp:5721]: Yeah, a full pass / 2 full passes are intentional, to actually allow the additional attack / 2 attacks as seems to be intended by the book. Part of the more generous reading is to try to get players to actually consider this 'ware as a feasible option.

Though maybe the potential for 2 more rounds of "out of initiative" messages outweighs that, so I'll tone that down.

**Review by luciensadi — APPROVED** [review]
LGTM, thanks!

**Pull request merged**